### DISCUSSION TITLE
Make it work with Jenkins?!? Just add the busybox-apk!

### DISCUSSION THREAD

**Discussion by koffeiniker** [squidfunk/mkdocs-material#4852] [Ideas]
Jenkins offers build pipeline with docker containers.

Regrettably the plugin is a bit nerdy and it requires the command "cat" to work. Easy to override the --entrypoint to sh, but cat is missing.

Could you integrate "apk add busybox", pretty please?

Until then it is always possible to integrate a Dockerfile to derive the image and adapt it. But just adding busybox and thus the cat command would do the trick ;-).

Thanks in advance!

--- Some background:
The docker_pipeline plugin of jenkins assumes to start the container on "cat", thus keeping it alive until it closes the stdin. It sideways executes the build steps in that containers using the equiv of "docker exec", I presume.

**Comment by squidfunk** [comment]
Thanks for suggesting. How much would it increase the size of the Docker image?

**Reply by koffeiniker** [reply]
Not checking in detail, it looks like roughly 2 MB. Sounds like quite a price for being able to be run by Jenkins. On the other hand it also offers a good deal of usual shell/text-tools just in case one really has to go hunting issues in the inner parts of the image.

```
REPOSITORY                    TAG         IMAGE ID       CREATED         SIZE
koffeiniker/material4mkdocs   latest      677109ba6c11   19 hours ago    139MB
squidfunk/mkdocs-material     latest      f521948bba58   3 days ago      137MB
````
To be able to run under Jenkins, then, the entrypoint needs to be adapted: "--entrypoint sh" which is a matter of mentioning in the documentation.

The resulting Jenkinsfile might look like this (mine is different, I still use a Dockerfile on the fly):
```
pipeline {

        agent {
                docker {
                        image 'koffeiniker/material4mkdocs'
                        args '-v $PWD:/docs --entrypoint sh'
                }
        }
        stages {
                stage('Build') {
                        steps {
                                sh 'mkdocs build'
                        }
                }
        }
}
```

**Reply by squidfunk** [reply]
Then I suggest you extend our image:

``` Dockerfile
FROM squidfunk/mkdocs-material
RUN apk add busybox
```

**Comment by koffeiniker** [comment]
That's what I do already. Ok, I keep doing that.
